### ISSUE TITLE
Webpack loader for .jsx issue

### ISSUE THREAD

**Issue by christianmemije** [learningequality/kolibri-exercise-perseus-plugin#105]
* Using 0.6 Kolibri and 0.7 perseus branch.
* Webpack is unable to find the appropriate loader to handle `.jsx` files

```
 ERROR in .-exercise-perseus-plugin/kolibri_exercise_perseus_plugin/~/perseus/src/widgets/simple-markdown-tester-editor.jsx
    Module parse failed: /home/christian/repos/kolibri-exercise-perseus-plugin/kolibri_exercise_perseus_plugin/node_modules/perseus/src/widgets/simple-markdown-tester-editor.jsx Unexpected token (17:15)
    You may need an appropriate loader to handle this file type.
```
etc..

**Comment by rtibbles** [comment]
Odd, I have specifically added a loader for this, and it has built fine for me.

See the webpack config for this plugin.

If you changed perseus renderer branches after starting the build watcher this could happen - is that the case?

**Comment by rtibbles** [comment]
Still not able to replicate this. Did you run `yarn install` after installing the plugin with `pip install -e ....`?

**Comment by christianmemije** [comment]
@rtibbles Still replicating. I also get the following errors when I run the `update_perseus.sh` script.

```
$ node ./extractPerseusMessages.js
Parsing error in file:  node_modules/perseus/src/article-diff-demo.jsx
Parsing error in file:  node_modules/perseus/src/item-diff-demo.jsx
Parsing error in file:  node_modules/perseus/src/renderer.jsx
Parsing error in file:  node_modules/perseus/src/widgets/passage.jsx
Parsing error in file:  node_modules/perseus/src/widgets/radio/choice-icon.jsx
Parsing error in file:  node_modules/perseus/src/widgets/interaction/arrow-picker.jsx
Parsing error in file:  node_modules/perseus/src/widgets/interaction/color-picker.jsx
Parsing error in file:  node_modules/perseus/src/widgets/interaction/dash-picker.jsx
Parsing error in file:  node_modules/perseus/src/widgets/interaction/element-container.jsx
Parsing error in file:  node_modules/perseus/src/multi-items/item-types.js
Parsing error in file:  node_modules/perseus/src/multi-items/items.js
[...ETC...]
```

**Comment by rtibbles** [comment]
Those errors are fine, I should probably suppress them, but there's rarely a need to run update_perseus.

What is a little odd, is that the `editor` files should not be being imported at all. Is it all jsx files that are throwing parsing errors, or just particular ones?

Also double check that this is running properly? `git submodule update --init --recursive`?

**Comment by christianmemije** [comment]
Quite a few, if not all.

`git submodule update --init --recursive` looks like it's working properly.

**Comment by rtibbles** [comment]
(When I say parsing errors, I mean webpack module parse failed errors, rather than the ones from the extractPerseusMessages)

Will keep thinking through what might cause it.

**Comment by christianmemije** [comment]
> (When I say parsing errors, I mean webpack module parse failed errors, rather than the ones from the extractPerseusMessages)

Yes, quite a few. 

It'd be best to see if anyone else can replicate it. @DXCanas @indirectlylit @jonboiser Can any of you  replicate this?

**Comment by indirectlylit** [comment]
What are the steps to reproduce?

I don't see any jsx error when I run the kolibri devserver on 0.6:

```
kolibri manage devserver --debug -- --webpack --lint
```

However, the `exercise_perseus_render_module` is not one of the child modules that get listed as having been compiled:

```
(le)~/P/l/r/kolibri (release-v0.6.x) $ kolibri manage devserver --debug -- --webpack --lint
DEBUG    Debug mode is on!
INFO     Starting webpack process with linting from Django runserver command
INFO     Django Runserver command has spawned a Webpack watcher process on pid 98904
yarn run v0.27.5
$ node ./frontend_build/src/webpackdevserver.js "--lint"
DEBUG    Debug mode is on!
Performing system checks...

System check identified no issues (1 silenced).
September 19, 2017 - 09:11:45
Django version 1.9.7, using settings 'kolibri.deployment.default.settings.base'
Starting development server at http://127.0.0.1:8000/
Quit the server with CONTROL-C.
WARNING  assets/src/module.js not found for plugin exercise_perseus_render_module.
WARNING:kolibri.core.webpack.hooks:assets/src/module.js not found for plugin exercise_perseus_render_module.
INFO     Writing webpack_json output to /var/folders/n9/k997qvqs3f7_m_310crwsxyc0000gp/T/117819-98910-ftrs2a.9a966tuik9.json
INFO:kolibri.core.webpack.management.commands.webpack_json:Writing webpack_json output to /var/folders/n9/k997qvqs3f7_m_310crwsxyc0000gp/T/117819-98910-ftrs2a.9a966tuik9.json
Hash: 234890eacf90ed28d1ebfcfc710c58b2531055ae7a30d00a6dfd2f64adda10d5b2b2a18b117d016d2cc89f9263688c63457881d3aab6a09fc58424580fc6ffe6922e7979754614d008835f7d56a823657bdeddb40e394da104622cc1181ddabb28c4af5c17d4a35a4fc76f8eb15f
Version: webpack 2.2.1
Child user_module:
    Hash: 234890eacf90ed28d1eb
    Time: 134936ms
                                              Asset     Size  Chunks                    Chunk Names
    background.png?748569b80a3a87b6dc5147547c151d53   428 kB          [emitted]  [big]  
          user_module-0.6.dev020170919090530-git.js   268 kB       0  [emitted]  [big]  user_module
          user_module0.6.dev020170919090530-git.css  35.3 kB       0  [emitted]         user_module
      user_module-0.6.dev020170919090530-git.js.map   360 kB       0  [emitted]         user_module
      user_module0.6.dev020170919090530-git.css.map  57.2 kB       0  [emitted]         user_module
      user_module0.6.dev020170919090530-git.rtl.css  35.4 kB       0  [emitted]         user_module
    Child extract-text-webpack-plugin:
                                                  Asset    Size  Chunks  Chunk Names
        background.png?748569b80a3a87b6dc5147547c151d53  428 kB          
Child style_guide_module:
    Hash: fcfc710c58b2531055ae
    Time: 80972ms
                                                   Asset     Size  Chunks                    Chunk Names
        style_guide_module-0.6.dev020170919090530-git.js  1.82 MB       0  [emitted]  [big]  style_guide_module
        style_guide_module0.6.dev020170919090530-git.css   9.6 kB       0  [emitted]         style_guide_module
    style_guide_module-0.6.dev020170919090530-git.js.map  2.24 MB       0  [emitted]         style_guide_module
    style_guide_module0.6.dev020170919090530-git.css.map  24.7 kB       0  [emitted]         style_guide_module
    style_guide_module0.6.dev020170919090530-git.rtl.css   9.6 kB       0  [emitted]         style_guide_module
Child coach_module:
    Hash: 7a30d00a6dfd2f64adda
    Time: 147565ms
                                             Asset     Size  Chunks                    Chunk Names
        coach_module-0.6.dev020170919090530-git.js   963 kB       0  [emitted]  [big]  coach_module
        coach_module0.6.dev020170919090530-git.css  55.6 kB       0  [emitted]         coach_module
    coach_module-0.6.dev020170919090530-git.js.map  1.29 MB       0  [emitted]         coach_module
    coach_module0.6.dev020170919090530-git.css.map   112 kB       0  [emitted]         coach_module
    coach_module0.6.dev020170919090530-git.rtl.css  55.7 kB       0  [emitted]         coach_module
Child setup_wizard:
    Hash: 10d5b2b2a18b117d016d
    Time: 121601ms
                                             Asset     Size  Chunks                    Chunk Names
        setup_wizard-0.6.dev020170919090530-git.js   873 kB       0  [emitted]  [big]  setup_wizard
        setup_wizard0.6.dev020170919090530-git.css  22.8 kB       0  [emitted]         setup_wizard
    setup_wizard-0.6.dev020170919090530-git.js.map  1.11 MB       0  [emitted]         setup_wizard
    setup_wizard0.6.dev020170919090530-git.css.map  40.8 kB       0  [emitted]         setup_wizard
    setup_wizard0.6.dev020170919090530-git.rtl.css  22.8 kB       0  [emitted]         setup_wizard
Child default_frontend:
    Hash: 2cc89f9263688c634578
    Time: 130910ms
                                                           Asset     Size  Chunks                    Chunk Names
                  default_frontend-0.6.dev020170919090530-git.js  1.58 MB       8  [emitted]  [big]  default_frontend
      MaterialIcons-Regular.ttf?a37b0c01c0baf1888ca812cc0508f6e2   128 kB          [emitted]         
    MaterialIcons-Regular.woff2?570eb83859dc23dd0eec423a49e147fe  44.3 kB          [emitted]         
            loading-spinner.gif?e850c29286b4eb51d1e5e00738b3fc04  98.8 kB          [emitted]         
                              intl-0.6.dev020170919090530-git.js   159 kB       0  [emitted]         intl
                           fakeRtl-0.6.dev020170919090530-git.js  46.7 kB       1  [emitted]         fakeRtl
                             sw-SW-0.6.dev020170919090530-git.js  24.7 kB       2  [emitted]         sw-SW
                             pt-PT-0.6.dev020170919090530-git.js  25.2 kB       3  [emitted]         pt-PT
                             fr-FR-0.6.dev020170919090530-git.js  24.7 kB       4  [emitted]         fr-FR
                                 5-0.6.dev020170919090530-git.js  24.5 kB       5  [emitted]         
                             es-ES-0.6.dev020170919090530-git.js  24.5 kB       6  [emitted]         es-ES
                                en-0.6.dev020170919090530-git.js  24.2 kB       7  [emitted]         en
     MaterialIcons-Regular.woff?012cf6a10129e2275d79d6adac7f3b02  57.6 kB          [emitted]         
                  default_frontend0.6.dev020170919090530-git.css   715 kB       8  [emitted]  [big]  default_frontend
                          intl-0.6.dev020170919090530-git.js.map   184 kB       0  [emitted]         intl
                       fakeRtl-0.6.dev020170919090530-git.js.map  63.9 kB       1  [emitted]         fakeRtl
                         sw-SW-0.6.dev020170919090530-git.js.map  29.5 kB       2  [emitted]         sw-SW
                         pt-PT-0.6.dev020170919090530-git.js.map  30.1 kB       3  [emitted]         pt-PT
                         fr-FR-0.6.dev020170919090530-git.js.map  29.5 kB       4  [emitted]         fr-FR
                             5-0.6.dev020170919090530-git.js.map  29.3 kB       5  [emitted]         
                         es-ES-0.6.dev020170919090530-git.js.map  29.3 kB       6  [emitted]         es-ES
                            en-0.6.dev020170919090530-git.js.map    29 kB       7  [emitted]         en
              default_frontend-0.6.dev020170919090530-git.js.map  2.06 MB       8  [emitted]         default_frontend
              default_frontend0.6.dev020170919090530-git.css.map   778 kB       8  [emitted]         default_frontend
              default_frontend0.6.dev020170919090530-git.rtl.css   715 kB       8  [emitted]  [big]  default_frontend
    
    WARNING in ./~/vue-loader/lib/template-compiler?{"id":"data-v-7a41f7d4","preserveWhitespace":false}!./~/svg-icon-inline-loader!./~/vue-loader/lib/selector.js?type=template&index=0!./~/htmlhint-loader?{"failOnError":true,"emitAs":"error"}!./kolibri/core/assets/src/views/side-nav/keen-menu-port/index.vue
    <menu-option v-for="option in options">: component lists rendered with v-for should have explicit keys. See https://vuejs.org/guide/list.html#key for more info.
     @ ./kolibri/core/assets/src/views/side-nav/keen-menu-port/index.vue 10:0-345
     @ ./~/buble-loader!./~/vue-loader/lib/selector.js?type=script&index=0!./~/htmlhint-loader?{"failOnError":true,"emitAs":"error"}!./kolibri/core/assets/src/views/side-nav/index.vue
     @ ./kolibri/core/assets/src/views/side-nav/index.vue
     @ ./kolibri/core/assets/src/core-app/apiSpec.js
     @ ./kolibri/core/assets/src/core-app/constructor.js
     @ ./kolibri/core/assets/src/core-app/index.js
    Child extract-text-webpack-plugin:
                                                               Asset     Size  Chunks  Chunk Names
          MaterialIcons-Regular.ttf?a37b0c01c0baf1888ca812cc0508f6e2   128 kB          
         MaterialIcons-Regular.woff?012cf6a10129e2275d79d6adac7f3b02  57.6 kB          
        MaterialIcons-Regular.woff2?570eb83859dc23dd0eec423a49e147fe  44.3 kB          
    Child extract-text-webpack-plugin:
                                                       Asset     Size  Chunks  Chunk Names
        loading-spinner.gif?e850c29286b4eb51d1e5e00738b3fc04  98.8 kB          
Child device_management_module:
    Hash: 81d3aab6a09fc5842458
    Time: 157338ms
                                                         Asset     Size  Chunks                    Chunk Names
        device_management_module-0.6.dev020170919090530-git.js   484 kB       0  [emitted]  [big]  device_management_module
        device_management_module0.6.dev020170919090530-git.css  18.6 kB       0  [emitted]         device_management_module
    device_management_module-0.6.dev020170919090530-git.js.map   634 kB       0  [emitted]         device_management_module
    device_management_module0.6.dev020170919090530-git.css.map  47.6 kB       0  [emitted]         device_management_module
    device_management_module0.6.dev020170919090530-git.rtl.css  18.6 kB       0  [emitted]         device_management_module
Child html5_app_renderer_module:
    Hash: 0fc6ffe6922e79797546
    Time: 67969ms
                                                          Asset       Size  Chunks             Chunk Names
        html5_app_renderer_module-0.6.dev020170919090530-git.js    21.1 kB       0  [emitted]  html5_app_renderer_module
        html5_app_renderer_module0.6.dev020170919090530-git.css  904 bytes       0  [emitted]  html5_app_renderer_module
    html5_app_renderer_module-0.6.dev020170919090530-git.js.map    25.9 kB       0  [emitted]  html5_app_renderer_module
    html5_app_renderer_module0.6.dev020170919090530-git.css.map    2.23 kB       0  [emitted]  html5_app_renderer_module
    html5_app_renderer_module0.6.dev020170919090530-git.rtl.css  902 bytes       0  [emitted]  html5_app_renderer_module
Child media_player_module:
    Hash: 14d008835f7d56a82365
    Time: 76565ms
                                                    Asset     Size  Chunks                    Chunk Names
             VideoJS.eot?9499bb60a03b7e136a699cd87b159cc0  7.25 kB          [emitted]         
             VideoJS.eot?00103e881a36640a08f869ff6888f0fd   5.9 kB          [emitted]         
        media_player_module-0.6.dev020170919090530-git.js   949 kB       0  [emitted]  [big]  media_player_module
        media_player_module0.6.dev020170919090530-git.css  68.1 kB       0  [emitted]         media_player_module
    media_player_module-0.6.dev020170919090530-git.js.map  1.17 MB       0  [emitted]         media_player_module
    media_player_module0.6.dev020170919090530-git.css.map   105 kB       0  [emitted]         media_player_module
    media_player_module0.6.dev020170919090530-git.rtl.css  68.1 kB       0  [emitted]         media_player_module
    Child extract-text-webpack-plugin:
                                               Asset     Size  Chunks  Chunk Names
        VideoJS.eot?9499bb60a03b7e136a699cd87b159cc0  7.25 kB          
        VideoJS.eot?00103e881a36640a08f869ff6888f0fd   5.9 kB          
Child learn_module:
    Hash: 7bdeddb40e394da10462
    Time: 153604ms
                                             Asset     Size  Chunks                    Chunk Names
        learn_module-0.6.dev020170919090530-git.js   817 kB       0  [emitted]  [big]  learn_module
        learn_module0.6.dev020170919090530-git.css  39.4 kB       0  [emitted]         learn_module
    learn_module-0.6.dev020170919090530-git.js.map  1.08 MB       0  [emitted]         learn_module
    learn_module0.6.dev020170919090530-git.css.map  84.2 kB       0  [emitted]         learn_module
    learn_module0.6.dev020170919090530-git.rtl.css  39.4 kB       0  [emitted]         learn_module
Child document_pdf_render_module:
    Hash: 2cc1181ddabb28c4af5c
    Time: 116215ms
                                                           Asset     Size  Chunks                    Chunk Names
                                 0-0.6.dev020170919090530-git.js  1.44 MB       0  [emitted]  [big]  
        document_pdf_render_module-0.6.dev020170919090530-git.js  1.58 MB       1  [emitted]  [big]  document_pdf_render_module
                       pdfJSWorker-0.6.dev020170919090530-git.js  1.45 MB    2, 0  [emitted]  [big]  pdfJSWorker
        document_pdf_render_module0.6.dev020170919090530-git.css  17.3 kB       1  [emitted]         document_pdf_render_module
                             0-0.6.dev020170919090530-git.js.map  1.72 MB       0  [emitted]         
    document_pdf_render_module-0.6.dev020170919090530-git.js.map  1.93 MB       1  [emitted]         document_pdf_render_module
    document_pdf_render_module0.6.dev020170919090530-git.css.map  27.1 kB       1  [emitted]         document_pdf_render_module
                   pdfJSWorker-0.6.dev020170919090530-git.js.map  1.73 MB    2, 0  [emitted]         pdfJSWorker
    document_pdf_render_module0.6.dev020170919090530-git.rtl.css  17.3 kB       1  [emitted]         document_pdf_render_module
Child management_module:
    Hash: 17d4a35a4fc76f8eb15f
    Time: 139394ms
                                                  Asset     Size  Chunks                    Chunk Names
        management_module-0.6.dev020170919090530-git.js   532 kB       0  [emitted]  [big]  management_module
        management_module0.6.dev020170919090530-git.css  36.2 kB       0  [emitted]         management_module
    management_module-0.6.dev020170919090530-git.js.map   707 kB       0  [emitted]         management_module
    management_module0.6.dev020170919090530-git.css.map  72.5 kB       0  [emitted]         management_module
    management_module0.6.dev020170919090530-git.rtl.css  36.2 kB       0  [emitted]         management_module
webpack: Compiled with warnings.
```

**Comment by christianmemije** [comment]
Pip install the 0.7 kolibri-exercise-perseus-plugin branch

https://github.com/learningequality/kolibri-exercise-perseus-plugin#how-can-i-install-this-plugin-for-development

@indirectlylit

**Comment by indirectlylit** [comment]
okay yes - I can reproduce.

After installing the perseus plugin for development, building kolibri front-end assets does give me the JSX loader errors. Stuff like this, many times over:

```
    ERROR in ../perseus-plugin/kolibri_exercise_perseus_plugin/~/perseus/src/article-renderer.jsx
    Module parse failed: /Users/d/Projects/learning_equality/repos/perseus-plugin/kolibri_exercise_perseus_plugin/node_modules/perseus/src/article-renderer.jsx Unexpected token (29:8)
    You may need an appropriate loader to handle this file type.
    | const ArticleRenderer = React.createClass({
    |     propTypes: {
    |         ...ProvideKeypad.propTypes,
    |         apiOptions: React.PropTypes.shape({
    |             onFocusChange: React.PropTypes.func,
     @ ../perseus-plugin/kolibri_exercise_perseus_plugin/~/perseus/src/perseus.js 14:24-57
     @ ./~/buble-loader!./~/vue-loader/lib/selector.js?type=script&index=0!./~/htmlhint-loader?{"failOnError":true,"emitAs":"error"}!../perseus-plugin/kolibri_exercise_perseus_plugin/assets/src/views/index.vue
     @ ../perseus-plugin/kolibri_exercise_perseus_plugin/assets/src/views/index.vue
     @ ../perseus-plugin/kolibri_exercise_perseus_plugin/assets/src/module.js
```

**Comment by rtibbles** [comment]
Hrm, confused as to what I have managed here. I cloned a new copy of the repo, and it still works for me.

**Comment by rtibbles** [comment]
Tried deleting my `node_modules` directory in kolibri also, in case a loader was lurking there, but still compiles just fine...

The only thing I can think of, is that for some reason the regex that is matching the `jsx` files is working on my machine, but not on anyone else's. But I can't imagine how that would be the case.

**Comment by indirectlylit** [comment]
did you try running `update_perseus.sh`?

**Comment by rtibbles** [comment]
Yes, ran through the whole thing. The parsing errors generated there are separate from the parsing errors in the webpack compilation though.

**Comment by DXCanas** [comment]
I'm actually reproducing too @rtibbles 

Didn't happen when I was doing `pip install .` (Maybe wasn't even installing it properly in this case?). But I'm now getting the error described above.

**Comment by DXCanas** [comment]
OK. Been trying to figure out what's going on for a while now, and I've narrowed it down a bit. 

Looks like the custom webpack config in the plugin isn't being taken into account because our build process can't find the file.

For whatever reason, 'parseBundlePlugin''s use of `require(path.resolve(path.join(data.plugin_path, 'webpack.config.js')))` fails inside of the perseus plugin's `data.plugin_path`. It's especially weird because if I do an `fs.readdirSync(data.plugin_path)` on the perseus plugin, it gets listed fine. See output below:

```
$ yarn run build
yarn run v0.27.5
$ webpack --config ./frontend_build/src/webpack.config.prod.js
INFO     Writing webpack_json output to /tmp/117821-16561-1pputml.xburgm0a4i.json
INFO:kolibri.core.webpack.management.commands.webpack_json:Writing webpack_json output to /tmp/117821-16561-1pputml.xburgm0a4i.json
Displaying info for: /home/david/Work/kolibri/kolibri/plugins/management
Path looks like: /n [ '__init__.py',
  '__pycache__',
  'assets',
  'build',
  'hooks.py',
  'kolibri_plugin.py',
  'static',
  'templates',
  'templatetags',
  'urls.py',
  'views.py' ]
Looking for:  /home/david/Work/kolibri/kolibri/plugins/management/webpack.config.js
Failed with: 
 { Error: Cannot find module '/home/david/Work/kolibri/kolibri/plugins/management/webpack.config.js'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at parseBundlePlugin (/home/david/Work/kolibri/frontend_build/src/parse_bundle_plugin.js:62:20)
    at readBundlePlugins (/home/david/Work/kolibri/frontend_build/src/read_bundle_plugins.js:56:18)
    at Object.<anonymous> (/home/david/Work/kolibri/frontend_build/src/webpack.config.js:19:18)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32) code: 'MODULE_NOT_FOUND' } 
 No local config


Displaying info for: /home/david/Work/kolibri/kolibri/plugins/setup_wizard
Path looks like: /n [ '__init__.py',
  '__pycache__',
  'assets',
  'build',
  'hooks.py',
  'kolibri_plugin.py',
  'middleware.py',
  'static',
  'templates',
  'templatetags',
  'urls.py',
  'views.py' ]
Looking for:  /home/david/Work/kolibri/kolibri/plugins/setup_wizard/webpack.config.js
Failed with: 
 { Error: Cannot find module '/home/david/Work/kolibri/kolibri/plugins/setup_wizard/webpack.config.js'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at parseBundlePlugin (/home/david/Work/kolibri/frontend_build/src/parse_bundle_plugin.js:62:20)
    at readBundlePlugins (/home/david/Work/kolibri/frontend_build/src/read_bundle_plugins.js:56:18)
    at Object.<anonymous> (/home/david/Work/kolibri/frontend_build/src/webpack.config.js:19:18)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32) code: 'MODULE_NOT_FOUND' } 
 No local config


Displaying info for: /home/david/Work/kolibri/kolibri/plugins/management
Path looks like: /n [ '__init__.py',
  '__pycache__',
  'assets',
  'build',
  'hooks.py',
  'kolibri_plugin.py',
  'static',
  'templates',
  'templatetags',
  'urls.py',
  'views.py' ]
Looking for:  /home/david/Work/kolibri/kolibri/plugins/management/webpack.config.js
Failed with: 
 { Error: Cannot find module '/home/david/Work/kolibri/kolibri/plugins/management/webpack.config.js'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at parseBundlePlugin (/home/david/Work/kolibri/frontend_build/src/parse_bundle_plugin.js:62:20)
    at readBundlePlugins (/home/david/Work/kolibri/frontend_build/src/read_bundle_plugins.js:56:18)
    at Object.<anonymous> (/home/david/Work/kolibri/frontend_build/src/webpack.config.js:19:18)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32) code: 'MODULE_NOT_FOUND' } 
 No local config


Displaying info for: /home/david/Work/kolibri/kolibri/plugins/learn
Path looks like: /n [ '__init__.py',
  '__pycache__',
  'assets',
  'build',
  'hooks.py',
  'kolibri_plugin.py',
  'node_modules',
  'static',
  'templates',
  'templatetags',
  'urls.py',
  'views.py' ]
Looking for:  /home/david/Work/kolibri/kolibri/plugins/learn/webpack.config.js
Failed with: 
 { Error: Cannot find module '/home/david/Work/kolibri/kolibri/plugins/learn/webpack.config.js'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at parseBundlePlugin (/home/david/Work/kolibri/frontend_build/src/parse_bundle_plugin.js:62:20)
    at readBundlePlugins (/home/david/Work/kolibri/frontend_build/src/read_bundle_plugins.js:56:18)
    at Object.<anonymous> (/home/david/Work/kolibri/frontend_build/src/webpack.config.js:19:18)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32) code: 'MODULE_NOT_FOUND' } 
 No local config


Displaying info for: /home/david/Work/kolibri/kolibri/plugins/style_guide
Path looks like: /n [ '__init__.py',
  '__pycache__',
  'assets',
  'build',
  'hooks.py',
  'kolibri_plugin.py',
  'node_modules',
  'package.json',
  'static',
  'templates',
  'templatetags',
  'urls.py',
  'views.py',
  'vue-doc-loader.js',
  'webpack.config.js',
  'yarn.lock' ]
Looking for:  /home/david/Work/kolibri/kolibri/plugins/style_guide/webpack.config.js
Found! Local Config is: 
 { resolveLoader: { alias: { 'vue-doc': '/home/david/Work/kolibri/kolibri/plugins/style_guide/vue-doc-loader' } } }


Displaying info for: /home/david/Work/perseus-plugin/kolibri_exercise_perseus_plugin
Path looks like: /n [ '.eslintrc.js',
  '__init__.py',
  '__pycache__',
  'assets',
  'build',
  'content_types.json',
  'extractPerseusMessages.js',
  'generateKATranslations.js',
  'getMessages.js',
  'gettextToICU.js',
  'kolibri_plugin.py',
  'locale',
  'node_modules',
  'npm-debug.log',
  'package.json',
  'static',
  'webpack.config.js',
  'yarn.lock' ]
Looking for:  /home/david/Work/perseus-plugin/kolibri_exercise_perseus_plugin/webpack.config.js
Failed with: 
 { Error: Cannot find module 'webpack'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/home/david/Work/perseus-plugin/kolibri_exercise_perseus_plugin/webpack.config.js:6:15)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3) code: 'MODULE_NOT_FOUND' } 
 No local config


Displaying info for: /home/david/Work/kolibri/kolibri/plugins/media_player
Path looks like: /n [ '__init__.py',
  '__pycache__',
  'assets',
  'build',
  'content_types.json',
  'kolibri_plugin.py',
  'node_modules',
  'package.json',
  'static',
  'yarn.lock' ]
Looking for:  /home/david/Work/kolibri/kolibri/plugins/media_player/webpack.config.js
Failed with: 
 { Error: Cannot find module '/home/david/Work/kolibri/kolibri/plugins/media_player/webpack.config.js'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at parseBundlePlugin (/home/david/Work/kolibri/frontend_build/src/parse_bundle_plugin.js:62:20)
    at readBundlePlugins (/home/david/Work/kolibri/frontend_build/src/read_bundle_plugins.js:56:18)
    at Object.<anonymous> (/home/david/Work/kolibri/frontend_build/src/webpack.config.js:19:18)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32) code: 'MODULE_NOT_FOUND' } 
 No local config


Displaying info for: /home/david/Work/kolibri/kolibri/plugins/user
Path looks like: /n [ '__init__.py',
  '__pycache__',
  'assets',
  'build',
  'hooks.py',
  'kolibri_plugin.py',
  'node_modules',
  'package.json',
  'static',
  'templates',
  'templatetags',
  'urls.py',
  'views.py',
  'yarn.lock' ]
Looking for:  /home/david/Work/kolibri/kolibri/plugins/user/webpack.config.js
Failed with: 
 { Error: Cannot find module '/home/david/Work/kolibri/kolibri/plugins/user/webpack.config.js'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at parseBundlePlugin (/home/david/Work/kolibri/frontend_build/src/parse_bundle_plugin.js:62:20)
    at readBundlePlugins (/home/david/Work/kolibri/frontend_build/src/read_bundle_plugins.js:56:18)
    at Object.<anonymous> (/home/david/Work/kolibri/frontend_build/src/webpack.config.js:19:18)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32) code: 'MODULE_NOT_FOUND' } 
 No local config


Displaying info for: /home/david/Work/kolibri/kolibri/plugins/coach
Path looks like: /n [ '__init__.py',
  '__pycache__',
  'api.py',
  'api_urls.py',
  'assets',
  'build',
  'hooks.py',
  'kolibri_plugin.py',
  'node_modules',
  'package.json',
  'serializers.py',
  'static',
  'templates',
  'templatetags',
  'test',
  'urls.py',
  'utils',
  'views.py',
  'yarn.lock' ]
Looking for:  /home/david/Work/kolibri/kolibri/plugins/coach/webpack.config.js
Failed with: 
 { Error: Cannot find module '/home/david/Work/kolibri/kolibri/plugins/coach/webpack.config.js'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at parseBundlePlugin (/home/david/Work/kolibri/frontend_build/src/parse_bundle_plugin.js:62:20)
    at readBundlePlugins (/home/david/Work/kolibri/frontend_build/src/read_bundle_plugins.js:56:18)
    at Object.<anonymous> (/home/david/Work/kolibri/frontend_build/src/webpack.config.js:19:18)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32) code: 'MODULE_NOT_FOUND' } 
 No local config


Displaying info for: /home/david/Work/kolibri/kolibri/plugins/html5_app_renderer
Path looks like: /n [ '__init__.py',
  '__pycache__',
  'assets',
  'build',
  'content_types.json',
  'kolibri_plugin.py',
  'node_modules',
  'package.json',
  'static',
  'yarn.lock' ]
Looking for:  /home/david/Work/kolibri/kolibri/plugins/html5_app_renderer/webpack.config.js
Failed with: 
 { Error: Cannot find module '/home/david/Work/kolibri/kolibri/plugins/html5_app_renderer/webpack.config.js'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at parseBundlePlugin (/home/david/Work/kolibri/frontend_build/src/parse_bundle_plugin.js:62:20)
    at readBundlePlugins (/home/david/Work/kolibri/frontend_build/src/read_bundle_plugins.js:56:18)
    at Object.<anonymous> (/home/david/Work/kolibri/frontend_build/src/webpack.config.js:19:18)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32) code: 'MODULE_NOT_FOUND' } 
 No local config


Displaying info for: /home/david/Work/kolibri/kolibri/core
Path looks like: /n [ '__init__.py',
  '__pycache__',
  'api.py',
  'apps.py',
  'assets',
  'build',
  'context_processors',
  'device',
  'discovery',
  'errors.py',
  'exams',
  'fields.py',
  'hooks.py',
  'kolibri_plugin.py',
  'models.py',
  'serializers.py',
  'settings.py',
  'static',
  'templates',
  'templatetags',
  'test',
  'urls.py',
  'views.py',
  'webpack' ]
Looking for:  /home/david/Work/kolibri/kolibri/core/webpack.config.js
Failed with: 
 { Error: Cannot find module '/home/david/Work/kolibri/kolibri/core/webpack.config.js'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at parseBundlePlugin (/home/david/Work/kolibri/frontend_build/src/parse_bundle_plugin.js:62:20)
    at readBundlePlugins (/home/david/Work/kolibri/frontend_build/src/read_bundle_plugins.js:56:18)
    at Object.<anonymous> (/home/david/Work/kolibri/frontend_build/src/webpack.config.js:19:18)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32) code: 'MODULE_NOT_FOUND' } 
 No local config


Displaying info for: /home/david/Work/kolibri/kolibri/plugins/document_pdf_render
Path looks like: /n [ '__init__.py',
  '__pycache__',
  'assets',
  'build',
  'content_types.json',
  'kolibri_plugin.py',
  'node_modules',
  'package.json',
  'static',
  'webpack.config.js',
  'yarn.lock' ]
Looking for:  /home/david/Work/kolibri/kolibri/plugins/document_pdf_render/webpack.config.js
Found! Local Config is: 
 { entry: { pdfJSWorker: 'pdfjs-dist/build/pdf.worker.entry' } }
```

I'm pretty stumped. Ideas? @rtibbles

**Comment by christianmemije** [comment]
Fixed in #107
